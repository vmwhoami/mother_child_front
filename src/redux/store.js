import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import ReduxThunk from 'redux-thunk';
import rootReducer from './rootReducer';

const store = createStore(rootReducer,
  composeWithDevTools(applyMiddleware(ReduxThunk)));

export default store;
