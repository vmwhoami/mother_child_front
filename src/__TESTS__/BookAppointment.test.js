import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import BookAppointment from '../components/pages/BookAppointmen';
import store from '../redux/store'
// import * as reactRedux from 'react-redux'

// jest.spyOn(reactRedux, 'useDispatch');


describe('Doctor', () => {
  test('snapshot renders correctly', () => {
    const component = renderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <BookAppointment />
        </Provider>
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});


describe('checks if Navbar component renders correctly', () => {
  test('should renders correct text', () => {
    const ren = render(
      <BrowserRouter>
        <Provider store={store}>
          <BookAppointment />
        </Provider>
      </BrowserRouter>,
    );
    console.log(ren);
    const text = screen.getByText('Full name');

    expect(text).toBeInTheDocument();
  });

  // test('should renders correct text', () => {
  //   render(
  //     <BrowserRouter>
  //       <Provider store={store}>
  //         <Home />
  //       </Provider>
  //     </BrowserRouter>,
  //   );
  //   const text = screen.getByText('Quote');
  //   expect(text).toBeInTheDocument();
  // });

  // test('should render correctly', () => {
  //   const tree = renderer
  //     .create(
  //       <BrowserRouter>
  //         <Provider store={store}>
  //           <Home />
  //         </Provider>
  //       </BrowserRouter>, ``,
  //     )
  //     .toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});