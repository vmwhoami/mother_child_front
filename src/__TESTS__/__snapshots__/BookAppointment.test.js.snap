// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Doctor snapshot renders correctly 1`] = `null`;

exports[`checks if Navbar component renders correctly should renders correct text 1`] = `null`;
